### ISSUE TITLE
PyMYSQL support

### ISSUE THREAD

**Issue by Andrew-Chen-Wang** [ClearcodeHQ/pytest-mysql#491]
### What action do you want to perform

We use PyMYSQL instead of mysqlclient. Is there a way to make mysqlclient optional as a dependency?

Also, would it be possible to add support in unit tests?

### What are the expected results

**Comment by fizyk** [comment]
1. Technically speaking, you can use both in your project, unless there's some issue that prevents you from installing mysqlclient as well?

The client library you use for code does not has to be the same test library uses to set up fixture data and read results. I usually encourage to not mix clients/sessions/connections between fixture/test code and the code being tested.

2. What do you mean support in unit tests?

**Comment by Andrew-Chen-Wang** [comment]
The current repo that I've been plumped into depends on PyMySQL and our unittests use sqlite, so I've been looking to potentially migrate to using MySQL for our test suite.

> I usually encourage to not mix clients/sessions/connections between fixture/test code and the code being tested.

Makes sense. Not sure if that's viable in my case (and honestly, not sure why lol), though internally it looks to be because we bar c extensions.

> What do you mean support in unit tests?

tl;dr ignore the comment: sorry, after forking the repo, I discovered that the package actually depended on mysqlclient directly, which I didn't know about when writing the first sentence. Had mysqlclient not actually be directly depended on by the package's code, I assumed the test matrix could easily just add pymysql in. Turns out, that's false.

**Comment by fizyk** [comment]
Technically speaking it would be possible, I had, for the time being, such a layer for pytest-postgresql, but there were only variants of psycopg2, which had same API.

Seeing as both libraries are in the same pymysql gothub org, I wonder if I shouldn't replace mysqlclient with pymysql. Especially that it's test library, not production, so speed is not that much of an issue, pymysql seems to be more popular 🤔

**Comment by Andrew-Chen-Wang** [comment]
thank you!
